core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -3887,62 +3890,56 @@ protected SqlBuilder getQueryString(boolean isGroupFiltering, boolean isUsername
             throw new UserStoreException("Condition is not valid.");
         }
 
+        SqlBuilder header = new SqlBuilder(new StringBuilder(sqlBuilder.getSql()));
+        addingWheres(sqlBuilder, header);
+
         for (ExpressionCondition expressionCondition : expressionConditions) {
-            if (ExpressionOperation.EQ.toString().equals(expressionCondition.getOperation()) && ExpressionAttribute
-                    .ROLE.toString().equals(expressionCondition.getAttributeName())) {
-                sqlBuilder.where("R.UM_ROLE_NAME = ?", expressionCondition.getAttributeValue());
+            if (ExpressionAttribute.ROLE.toString().equals(expressionCondition.getAttributeName())) {
+                if (!MYSQL.equals(dbType)) {
+                    multiGroupQueryBuilder(sqlBuilder, header, hitGroupFilter, expressionCondition);
+                    hitGroupFilter = true;
+                } else {
+                    //TODO
```

Comment:
Are we tracking this with a Git issue?

Reply:
fixed with later PR, also we track it. https://github.com/wso2/carbon-kernel/pull/1839